eng/pipelines/doc-index.yml

```diff
@@ -11,40 +11,63 @@ jobs:
-        workingDirectory: $(Build.SourcesDirectory)
+          echo "##vso[task.setvariable variable=docfxPath]$(Build.BinariesDirectory)/docfx/docfx.exe"
+        workingDirectory: $(Build.BinariesDirectory)
         displayName: Download and Extract DocFX
-        name: setupDocfxTool
       - task: PowerShell@2
         displayName: "Generate Doc Index"
         inputs:
           pwsh: true
           filePath: $(Build.SourcesDirectory)/eng/docs/index/Generate-DocIndex.ps1
           arguments: >
-            -Docfx $(setupDocfxTool.docfxPath)
+            -Docfx $(docfxPath)
             -RepoRoot $(Build.SourcesDirectory)
             -DocGenDir "$(Build.SourcesDirectory)/eng/docs/index"
+            -DocOutDir $(Build.ArtifactStagingDirectory)/docfx_project
             -verbose
-
       - task: UsePythonVersion@0
         displayName: 'Use Python 3.6'
         inputs:
           versionSpec: '3.6'
 
       - template: /eng/common/pipelines/templates/steps/mashup-doc-index.yml
         parameters:
-          SourceDirectory: $(Build.SourcesDirectory)
+          SourceDirectory: $(Build.ArtifactStagingDirectory)
 
-      - task: CopyFiles@2
-        displayName: Copy HTML to Artifacts Directory
-        inputs:
-          sourceFolder: $(Build.SourcesDirectory)/docfx_project/
-          content: '**\*'
-          targetFolder: $(Build.ArtifactStagingDirectory)/docfx_project
-          overWrite: true
+      - pwsh: |
+          Copy-Item -Path $(Build.SourcesDirectory)/eng/common/scripts/git-branch-push.ps1 -Destination ./ -Recurse -Force
+          echo "##vso[task.setvariable variable=gitPushScriptPath]$(Build.BinariesDirectory)/git-branch-push.ps1"
+        workingDirectory: $(Build.BinariesDirectory)
+        displayName: Move Git Push scripts to Tool Directory 
 
       - task: PublishPipelineArtifact@0
         condition: succeeded()
         inputs:
           artifactName: "Doc.Index"
-          targetPath: $(Build.ArtifactStagingDirectory)/docfx_project/_site
\ No newline at end of file
+          targetPath: $(Build.ArtifactStagingDirectory)/docfx_project/_site
+
+      - pwsh: |
+          git checkout -b gh-pages-local --track origin/gh-pages-root
+        workingDirectory: $(Build.SourcesDirectory)
+        displayName: Git pull GH pages branch
+
+      - pwsh: |
+          Copy-Item -Path $(Build.ArtifactStagingDirectory)/docfx_project/_site/* -Destination ./ -Recurse -Force
+          git add -A
+          git status
+        workingDirectory: $(Build.SourcesDirectory)
+        displayName: Copy the latest changes
+
+      - task: PowerShell@2
+        displayName: Push the Docs to GH-Pages
+        condition: succeeded()
+        inputs:
+          pwsh: true
+          workingDirectory: $(Build.SourcesDirectory)
+          filePath: $(gitPushScriptPath)
+          arguments: >
+            -PRBranchName "gh-pages1"
```

Comment:
This is fine testing but you will want this to be gh-pages once testing is done.